File: Linked_Lists/Medium/61/61.c
// Definition for singly-linked list.
struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode* rotateRight(struct ListNode* head, int k) {
  if(!head || !head->next){
    return head;
  }

  struct ListNode *tail = head;
  struct ListNode *temp = head;
  int length = 0;

  while(tail->next != NULL){
    tail = tail->next;
    length++;
  }

  if(k > length){
    k = k % (length + 1);
  }

  for(int i = 0; i < (length - k); i++){
    temp = temp->next;
  }

  tail->next = head;
  head = temp->next;
  temp->next = NULL;

  return head;
}

File: Linked_Lists/Medium/61/61.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
  def rotateRight(self, head, k):
    """
    :type head: Optional[ListNode]
    :type k: int
    :rtype: Optional[ListNode]
    """
    if head == None or head.next == None:
      return head

    tail = head
    temp = head
    length = 0
    while tail.next != None:
      tail = tail.next
      length += 1

    if k > length:
      k = k % (length + 1)

    for i in range(length - k):
      temp = temp.next

    tail.next = head
    head = temp.next
    temp.next = None

    return head